challenge-manager/edge-tracker/tracker.go

```diff
@@ -731,11 +727,12 @@ func (et *Tracker) openSubchallengeLeaf(ctx context.Context) error {
 		endParentCommitment, err = et.stateProvider.HistoryCommitment(
 			ctx,
 			&l2stateprovider.HistoryCommitmentRequest{
-				WasmModuleRoot:              et.wasmModuleRoot,
-				Batch:                       l2stateprovider.Batch(et.heightConfig.InboxMaxCount),
+				WasmModuleRoot:              et.associatedAssertionMetadata.WasmModuleRoot,
+				FromBatch:                   et.associatedAssertionMetadata.FromBatch,
+				ToBatch:                     et.associatedAssertionMetadata.ToBatch,
 				UpperChallengeOriginHeights: []l2stateprovider.Height{},
-				FromHeight:                  l2stateprovider.Height(et.heightConfig.StartBlockHeight),
```

Comment:
This was previously incorrect. Parent commitments in subchallenges should be from 0 --> N, and 0 --> N + 1